spec/javascripts/components/option-select-spec.js

```diff
@@ -349,110 +349,133 @@ describe('An option select component', function () {
     it('filters the checkboxes and updates the filter count correctly', function () {
       expect($('.govuk-checkboxes__item:visible').length).toBe(12)
 
-      $filterInput.val('in').keyup()
+      $filterInput.value = 'in'
+      window.GOVUK.triggerEvent($filterInput, 'keyup')
+
       jasmine.clock().tick(400)
       expect($('.govuk-checkboxes__input:visible').length).toBe(5)
       expect($count.text()).toBe('5 options found, 0 selected')
 
-      $filterInput.val('ind').keyup()
+      $filterInput.value = 'ind'
+      window.GOVUK.triggerEvent($filterInput, 'keyup')
       jasmine.clock().tick(400)
       expect($('.govuk-checkboxes__input:visible').length).toBe(2)
       expect($count.html()).toBe('2 options found, 0 selected')
 
-      $filterInput.val('shouldnotmatchanything').keyup()
+      $filterInput.value = 'shouldnotmatchanything'
+      window.GOVUK.triggerEvent($filterInput, 'keyup')
       jasmine.clock().tick(400)
       expect($('.govuk-checkboxes__input:visible').length).toBe(0)
       expect($count.html()).toBe('0 options found, 0 selected')
     })
 
     it('does not propagate keypresses up', function () {
-      var e = jQuery.Event('keyup', { keyCode: 13 }) // enter
-      $filterInput.trigger(e)
+      var events = document.createEvent("Events")
+      events.initEvent('keyup', true, true)
+      events.keyCode = 13;
+      events.which = 13;
+      events.charCode = 13;
+      events.key = 'Enter';
+      events.code = 'Enter';
+      $filterInput.dispatchEvent(events)
```

Comment:
This approach works ok in IE 8, but may not work in later versions.  @andysellick has a draft PR that adds the ability to pass params to `window.GOVUK.triggerEvent` so it can handle specific keys with `keyup` events.  When this PR is merged, this block should be updated to use that method.

Reply:
https://github.com/alphagov/govuk_publishing_components/pull/2090/commits/dc0a9f74d069db3d3a7ca50932c345c1b951b8a8 for reference.